docs/examples/rag_azuresearch.ipynb

```diff
@@ -0,0 +1,703 @@
+       "</pre>\n"
+      ],
+      "text/plain": [
+       "Uploaded batch \u001b[1;36m500\u001b[0m -> \u001b[1;36m750\u001b[0m; success: \u001b[3;92mTrue\u001b[0m, status code: \u001b[1;36m201\u001b[0m\n"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "text/html": [
+       "<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\">Uploaded batch <span style=\"color: #008080; text-decoration-color: #008080; font-weight: bold\">750</span> -&gt; <span style=\"color: #008080; text-decoration-color: #008080; font-weight: bold\">966</span>; success: <span style=\"color: #00ff00; text-decoration-color: #00ff00; font-style: italic\">True</span>, status code: <span style=\"color: #008080; text-decoration-color: #008080; font-weight: bold\">201</span>\n",
+       "</pre>\n"
+      ],
+      "text/plain": [
+       "Uploaded batch \u001b[1;36m750\u001b[0m -> \u001b[1;36m966\u001b[0m; success: \u001b[3;92mTrue\u001b[0m, status code: \u001b[1;36m201\u001b[0m\n"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "text/html": [
+       "<pre style=\"white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace\">All chunks uploaded to Azure Search.\n",
+       "</pre>\n"
+      ],
+      "text/plain": [
+       "All chunks uploaded to Azure Search.\n"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "from openai import AzureOpenAI\n",
+    "from azure.search.documents import SearchClient\n",
+    "import uuid\n",
+    "\n",
+    "search_client = SearchClient(AZURE_SEARCH_ENDPOINT, AZURE_SEARCH_INDEX_NAME, AzureKeyCredential(AZURE_SEARCH_KEY))\n",
+    "openai_client = AzureOpenAI(\n",
+    "    api_key=AZURE_OPENAI_API_KEY,\n",
+    "    api_version=AZURE_OPENAI_API_VERSION,\n",
+    "    azure_endpoint=AZURE_OPENAI_ENDPOINT,\n",
+    ")\n",
+    "\n",
+    "def embed_text(text: str):\n",
+    "    response = openai_client.embeddings.create(\n",
+    "        input=text,\n",
+    "        model=AZURE_OPENAI_EMBEDDINGS\n",
+    "    )\n",
+    "    return response.data[0].embedding\n",
+    "\n",
+    "upload_docs = []\n",
+    "for chunk_id, chunk_text in all_chunks:\n",
+    "    embedding_vector = embed_text(chunk_text)\n",
+    "    upload_docs.append(\n",
+    "        {\n",
+    "            \"chunk_id\": str(uuid.uuid4()),\n",
```

Comment:
In a previous cell you created a list of tuples (chunk id, chunk text), but here you're not using the id and generating a GUID instead. The previously generated chunk id doesn't seem to be used, seems you only need one thing, either the chunk ids you generated before (then you don't need the new GUID here, use the first element of the tuple instead), or don't generate chunk ids in the previous cell (and in that case it looks like you don't need to create all_chunks at all, since you could just enumerate the original doc_chunks instead.

Reply:
Updated to use the chunk_id first element of the tuple.